spark_log_parser/loaders/s3.py

```diff
@@ -5,24 +5,27 @@
 from urllib.parse import ParseResult, urlparse
 
 from botocore.client import BaseClient
+from botocore.response import StreamingBody
 
-from spark_log_parser.loaders import AbstractFileDataLoader, BlobFileReaderMixin, LinesFileReaderMixin
+from spark_log_parser.loaders import AbstractFileDataLoader, BlobFileReaderMixin, LinesFileReaderMixin, \
+    FileChunkStreamWrapper
 
+# boto3 clients are threadsafe, so we can use a singleton for all instances
+S3_CLIENT = boto3.client("s3")
```

Comment:
Seems we should be able to reuse this reliably across object instances - is there any reason we wouldn't want to that I'm missing?